android/src/main/java/com/stripeterminalreactnative/listener/RNReaderReconnectionListener.kt

```diff
@@ -0,0 +1,32 @@
+package com.stripeterminalreactnative.listener
+
+import com.facebook.react.bridge.ReactApplicationContext
+import com.stripe.stripeterminal.external.callable.Cancelable
+import com.stripe.stripeterminal.external.callable.ReaderReconnectionListener
+import com.stripe.stripeterminal.external.models.Reader
+import com.stripeterminalreactnative.ReactExtensions.sendEvent
+import com.stripeterminalreactnative.ReactNativeConstants.*
+import com.stripeterminalreactnative.mapFromReader
+
+class RNReaderReconnectionListener(
+    private val context: ReactApplicationContext
+) : ReaderReconnectionListener {
+
+    override fun onReaderReconnectFailed(reader: Reader) {
+        context.sendEvent(TERMINAL_FAIL_READER_RECONNECT.listenerName) {
+            putMap("reader", mapFromReader(reader))
+        }
+    }
+
+    override fun onReaderReconnectStarted(reader: Reader, cancelReconnect: Cancelable) {
```

Comment:
need to deal with the `Cancelable` here and expose a cancelReaderReconnect method similar to Android. Existing example: https://github.com/stripe/stripe-terminal-react-native/blob/1870117c/android/src/main/java/com/stripeterminalreactnative/listener/RNBluetoothReaderListener.kt#L35-L37

cc @chr-stripe